### PR TITLE
Update html-end-tags doc

### LINKED ISSUE #160: Question: `vue/html-end-tags`
**Tell us about your environment**

* **ESLint Version:** 4.4.1
* **eslint-plugin-vue Version:** 3.11.0
* **Node Version:** 8.3.0

**Question**

For now rule `vue/html-end-tags` works only for non void html tags as described in doc: 
https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/html-end-tags.md#todo-brbr
```
TODO: <br></br>

parse5 does not recognize the illegal end tags of void elements.
```

We are not longer use `parse5` but issue is stil present with new parser.

### PR DESCRIPTION
**Pull request by michalsnik**
Fixes #160

### CODE REVIEW

**Comment by michalsnik** [comment]
@mysticatea up

File: docs/rules/html-end-tags.md [hunk 1]
```diff
@@ -2,17 +2,12 @@
...
-- Other elements require end tags.
-
 ## :book: Rule Details
 
-This rule reports the following elements:
+This rule reports:
 
-- [Void elements] which have end tags.
-- Other elements which do not have end tags and are not self-closing.
+- presence of end tags on [Void elements](https://www.w3.org/TR/html51/syntax.html#void-elements)
```
Comment (chrisvfritz) [inline docs/rules/html-end-tags.md]: Maybe better to say "an and tag"?

File: docs/rules/html-end-tags.md [hunk 2]
```diff
@@ -2,17 +2,12 @@
...
+- absence of end tags or self-closed endings on other elements
```
Comment (chrisvfritz) [inline docs/rules/html-end-tags.md]: Might be clearer to say:

> absence of both an end tag (e.g. `</div>`) and self-closing opening tag (e.g. `<div/>`) on other elements

**Comment by michalsnik** [comment]
@chrisvfritz Thanks for suggestions! Updated :)

**Pull request merged**